Plugin/CheckElasticSearchAvailabilityPlugin.php

```diff
@@ -0,0 +1,57 @@
+<?php
+/**
+ * @package   Snowdog\AlpacaGeneral
+ * @copyright 2021 Snowdog. All rights reserved.
+ * @see       https://snow.dog/
+ */
+
+declare(strict_types=1);
+
+namespace Snowdog\AlpacaGeneral\Plugin;
+
+use Smile\ElasticsuiteCore\Index\IndexOperation;
+use Psr\Log\LoggerInterface;
+use Magento\Sales\Model\Service\OrderService;
+use Magento\Sales\Api\Data\OrderInterface;
+use Magento\Framework\Exception\LocalizedException;
+
+class CheckElasticSearchAvailabilityPlugin
+{
+    /**
+     * @var IndexOperation
+     */
+    private $indexOperation;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    public function __construct(
+        IndexOperation $indexOperation,
+        LoggerInterface $logger
+    ) {
+        $this->indexOperation = $indexOperation;
+        $this->logger = $logger;
+    }
+
+    /**
+     * @throws LocalizedException
+     */
+    public function aroundPlace(
```

Comment:
Please change it to `before` plugin type and please return the correct parameters

Reply:
done